...JAVASCRIPT_wrap.cxx:238:10: fatal error: 'uv.h' file not found

Hi, I am trying out current main branch (as of 15/07/2024). I am trying your fork to get smart pointers support. It seems there's a strong libuv dependency, which I am trying to sort out. I decided to go with [cmake-js](https://github.com/cmake-js/cmake-js) build and I am using [UseSWIG](https://cmake.org/cmake/help/latest/module/UseSWIG.html) macros. I use `-c++ -javascript -napi` args. I am getting the following error:
```
/build/CMakeFiles/rpc.dir/rpcJAVASCRIPT_wrap.cxx:238:10: fatal error: 'uv.h' file not found
#include <uv.h>
         ^~~~~~
1 error generated.
make[2]: *** [CMakeFiles/rpc.dir/CMakeFiles/rpc.dir/rpcJAVASCRIPT_wrap.cxx.o] Error 1
make[1]: *** [CMakeFiles/rpc.dir/all] Error 2
make: *** [all] Error 2
```
What is the best approach to get pass this error? Should I include emnapi?

--- a/CHANGES-JSE
+++ b/CHANGES-JSE
@@ -7,6 +7,10 @@ the issue number to the end of the URL: https://github.com/swig/swig/issues/
 Version 5.0.4
 ===========================
 
+2024-07-17: mmomtchev
+            [JavaScript] Fix [mmomtchev/swig#58](https://github.com/mmomtchev/swig/issues/58),
+            remove the unused dependency on the `uv.h` header
+
 2024-07-14: mmomtchev
             [JavaScript] Support emsdk up to 3.1.63
 

--- a/Lib/javascript/napi/javascriptruntime.swg
+++ b/Lib/javascript/napi/javascriptruntime.swg
@@ -32,7 +32,6 @@
 #error NAPI_VERSION 6 is the minimum supported target (Node.js >=14, >=12.17, >=10.20)
 #endif
 #include <napi.h>
-#include <uv.h>
 
 #include <errno.h>
 #include <limits.h>

